### PR TITLE
Support flag conditions based on Vespa release 

### PR DESCRIPTION
**Pull request by hakonhall**
Supports a "relational" condition with a new dimension "vespa-version", that
can be satisfied with e.g. "predicate": ">= 7.120.5" as long as the condition is
evaluated in a JVM that has a Vtag at least high as 7.120.5.

The typical use-case for this condition would be: The developer has used the
flag to test and verify the feature is ready to roll out globally. The
developer can now roll the feature with the next release, and ORCHESTRATED,
halting if anything goes wrong like any normal rollout.

This also allows one-shot tests of a feature flag in integration tests: Just
enable it for an upcoming version with predicate "== 7.x.y".

### CODE REVIEW

File: flags/src/main/java/com/yahoo/vespa/flags/json/RelationalCondition.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * @author hakonhall
+ */
+public class RelationalCondition implements Condition {
+    private final RelationalPredicate relationalPredicate;
+    private final Predicate<String> predicate;
+    private final FetchVector.Dimension dimension;
+
+    public static RelationalCondition create(CreateParams params) {
```
Comment (freva) [inline flags/src/main/java/com/yahoo/vespa/flags/json/RelationalCondition.java]: Make this into constructor like the others/change the others to use factory method?

File: flags/src/main/java/com/yahoo/vespa/flags/json/Condition.java [hunk 2]
```diff
@@ -1,62 +1,72 @@
...
+    class CreateParams {
+        private final FetchVector.Dimension dimension;
+        private final List<String> values;
+        private final Optional<String> predicate;
 
-        switch (type) {
-            case WHITELIST: return isMember;
-            case BLACKLIST: return !isMember;
-            default: throw new IllegalArgumentException("Unknown type " + type);
+        public CreateParams(FetchVector.Dimension dimension, List<String> values, Optional<String> predicate) {
```
Comment (freva) [inline flags/src/main/java/com/yahoo/vespa/flags/json/Condition.java]: Error when setting a predicate for whitelist/blacklist or setting values for relational?

**Review by freva — APPROVED** [review]

File: flags/src/main/java/com/yahoo/vespa/flags/FetchVector.java [hunk 3]
```diff
@@ -44,7 +34,27 @@
...
+         *
+         * <p>ALL flags can be set differently in different zones: This dimension is ONLY useful for the controller
+         * that needs to handle multiple zones.
+         *
+         * <p>Value from ZoneId::value is of the form environment.region.
+         */
+        ZONE_ID,
+
+        /**
+         * WARNING: DO SET THIS DIMENSION FOR A FLAG
```
[the remark this answers is not in the document]
Comment (hakonhall) [reply flags/src/main/java/com/yahoo/vespa/flags/FetchVector.java]: I have also documented HOSTNAME and toned down the comments.

File: flags/src/main/java/com/yahoo/vespa/flags/json/RelationalCondition.java (hunk 1 above)
Comment (hakonhall) [reply flags/src/main/java/com/yahoo/vespa/flags/json/RelationalCondition.java]: Made factories.

File: flags/src/main/java/com/yahoo/vespa/flags/json/Condition.java (hunk 2 above)
Comment (hakonhall) [reply flags/src/main/java/com/yahoo/vespa/flags/json/Condition.java]: Done

**Review by freva — APPROVED** [review]
:100:

**Pull request merged**